docs/agentpool.md

```diff
@@ -0,0 +1,29 @@
+# `AgentPool`
+
+`AgentPool` controller allows managing Terraform Cloud Agent Pools via Kubernetes Custom Resources.
```

Comment:
I would add a small clarification statement that this doesn't actually manage the lifecycle of the agent pool nodes at this point.

Reply:
Done, thank you!